cmd/podman/generate_kube.go

```diff
@@ -18,16 +19,15 @@ var (
 			Name:  "service, s",
 			Usage: "only generate YAML for kubernetes service object",
 		},
-		LatestFlag,
 	}
 	containerKubeDescription = "Generate Kubernetes Pod YAML"
 	containerKubeCommand     = cli.Command{
-		Name:                   "generate",
-		Usage:                  "Generate Kubernetes pod YAML for a container",
+		Name:                   "kube",
+		Usage:                  "Generate Kubernetes pod YAML for a container or pod",
 		Description:            containerKubeDescription,
 		Flags:                  sortFlags(containerKubeFlags),
 		Action:                 generateKubeYAMLCmd,
-		ArgsUsage:              "CONTAINER-NAME",
+		ArgsUsage:              "CONTAINER|POD-NAME",
```

Comment:
I'd make it explicit whether it is a container or a pod (i.e. add a `--pod` flag if we decide to use containers by default, or `--container` if the other way around).

Otherwise how to pick the pod if we have a pod and a container with the same name?

Reply:
That's actually not allowed by design - containers and pods share the same pool of names, so you can't have a container and pod with the same name.